src/inmanta/agent/executor.py

```diff
@@ -289,27 +404,137 @@ def initialize_envs_directory() -> str:
     return env_dir
 
 
-class VirtualEnvironmentManager:
+class PoolManager:
+    def __init__(self, retention_time: int, thread_pool: concurrent.futures.thread.ThreadPoolExecutor) -> None:
+        self.running = False
+        self._locks: inmanta.util.NamedLock = inmanta.util.NamedLock()
+
+        # We keep a reference to the periodic cleanup task to prevent it
+        # from disappearing mid-execution https://docs.python.org/3.11/library/asyncio-task.html#creating-tasks
+        self.cleanup_job: typing.Optional[asyncio.Task[None]] = None
+
+        self.retention_time: int = retention_time
+        self.thread_pool = thread_pool
+
+    async def start(self) -> None:
+        """
+        Start the cleaning job of the Pool Manager
+        """
+        if self.running:
+            raise RuntimeError(f"This manager `{self.__class__.__name__}` is already running!")
+        self.running = True
+        self.cleanup_job = asyncio.create_task(self.cleanup_inactive_pool_members_task())
+
+    async def stop(self) -> None:
+        """
+        Cancel the cleaning job of the Pool Manager -> We don't want to wait for the cleaning to finish because it could take a
+        long time.
+        """
+        self.running = False
+        if self.cleanup_job is not None:
+            try:
+                self.cleanup_job.cancel()
```

Comment:
Why do we have both running=False and also cancel? 

They both terminate the run, but differently?